### PR TITLE
feat(workflows): update MDN urls in mdn-content-inventory Dependabot PR

### PR DESCRIPTION
**Pull request by caugner**
#### Summary

Adds a workflow that runs on the Dependabot PRs for `ddbeck/mdn-content-inventory`, updates the MDN urls, and pushes the updates to the PR.

#### Test results and supporting details

Used the [update-browser-releases](https://github.com/mdn/browser-compat-data/blob/main/.github/workflows/update-browser-releases.yml) workflow as a template.

#### Related issues

Follow-up of https://github.com/mdn/browser-compat-data/pull/24669.
Unblocks https://github.com/mdn/browser-compat-data/pull/23810.

### CODE REVIEW

File: .github/workflows/update-mdn-urls.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+  schedule:
+    - cron: "30 4 * * 1-5"
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  update:
```
Comment (queengooborg) [inline .github/workflows/update-mdn-urls.yml:14]:
```suggestion
  update-mdn-urls:
```

**Comment by queengooborg** [comment]
This workflow would also need to update Daniel's package to get the latest list of pages.  Could we add that step in?

**Comment by caugner** [comment]
> This workflow would also need to update Daniel's package to get the latest list of pages. Could we add that step in?

I don't think they have to happen together, and I wouldn't want to couple them. Daniel's package would get auto-upgraded once we merge your PR that dowgrades these errors to warnings.

**Comment by Elchi3** [comment]
fyi, this workflow adds a comment to a dependabot PR: https://github.com/web-platform-dx/web-features/blob/main/.github/workflows/bcd_upgrade_checklist.yml

Maybe we could implement something similar here to push a commit to the `@ddbeck/mdn-content-inventory` dependabot PR?

**Comment by caugner** [comment]
> Maybe we could implement something similar here to push a commit to the `@ddbeck/mdn-content-inventory` dependabot PR?

Yes, this is what I have done now. 👍

File: .github/workflows/update-mdn-urls.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+name: Update MDN urls
+
+on:
+  push:
+    branches:
+      - "dependabot/npm_and_yarn/ddbeck/mdn-content-inventory-*"
```
Comment (caugner) [inline .github/workflows/update-mdn-urls.yml:6]: Only runs on the Dependabot PRs.

File: .github/workflows/update-mdn-urls.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+env:
+  GH_TOKEN: ${{ secrets.GH_TOKEN }}
```
Comment (caugner) [inline .github/workflows/update-mdn-urls.yml:9]: Authenticates as mdn-bot.

File: .github/workflows/update-mdn-urls.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: ".nvmrc"
+          cache: npm
+
+      - name: "Setup git"
+        run: |
+          git remote set-url origin https://mdn-bot:$GH_TOKEN@github.com/${{ github.repository }}.git
+          git config user.email 108879845+mdn-bot@users.noreply.github.com
+          git config user.name mdn-bot
```
Comment (caugner) [inline .github/workflows/update-mdn-urls.yml:31]: Commits and pushes as mdn-bot.

File: .github/workflows/update-mdn-urls.yml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+      - name: Install
+        run: npm ci
+
+      - name: Update
+        run: |
+          npm run lint:fix -- --only=mdn_urls
```
Comment (caugner) [inline .github/workflows/update-mdn-urls.yml:38]: We only fix `mdn_urls` here, ...

File: .github/workflows/update-mdn-urls.yml [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+          if git diff --exit-code; then
+            echo "No changes."
+          else
+            git commit -m 'chore: fix mdn urls' .
```
Comment (caugner) [inline .github/workflows/update-mdn-urls.yml:42]: ... but lint-staged will implicitly `lint:fix` everything in the changed files.

**Review by Elchi3 — APPROVED** [review]
Very nice! This reads good to me. I haven't actually tested it but feel free to merge and then we can see if it does the job with the next dependabot PR :)

**Pull request merged**